connect_four.py

```diff
@@ -97,77 +103,119 @@ def __diagonal_pos_slope(self, row: int, col: int, color: str) -> bool:
                 break
         return up_right + down_left >= self.win_req
 
-    def check_win(self, row: int, col: int, color: str) -> bool:
+    def check_win(self, col: int, color: str) -> bool:
+        row = (self.num_rows - len(self.columns[col]))
         return  self.__vertical_win(row, col, color) or self.__horizontal_win(row, col, color) or self.__diagonal_neg_slope(row, col, color) or self.__diagonal_pos_slope(row, col, color)
 
 
 
 # Main gameplay. Set as while loop to play until exited.
-class GameLoop:
-    #TODO split up into smaller funcitons 
+class GameLoop: 
     
     def __init__(self) -> None:
+        self.play = True
         self.winner = None
+        self.player1 = None
+        self.player2 = None
+        #self.cols, self.rows, self.win_len = self.game_setup()
         self.board = ConnectFourBoard()  #TODO add capability to change board size and win requirement
 
-    def __game_won(self, player: str):
-        self.board.print_board()
-        print(f"{player}, you win!! Press R to restart, or X to exit the game.")
-        response = str(input()).capitalize() 
+    '''def game_setup() -> int:
```

Comment:
hm why is this commented out?

Reply:
So I started working on this method, which is for asking the user to define the board size and win requirement, but I left it out bc it required a lot of IO logic. I will flesh it out more in the IO class I'll make for the next PR!